### ISSUE TITLE
Pharo does not work on network drive on Windows

### ISSUE THREAD

**Issue by JanBliznicenko** [guillep/pharo-core#10085]
**Bug description**
When Pharo VM is run from network drive, it is not able to find any file, including .changes, .sources, .dll etc.
This is an issue on company/university systems with user homes on network drives.
Same files work properly when copied to C drive or flash drive (on the same PC).

![Pharo3](https://user-images.githubusercontent.com/9571633/135471583-46eea282-c365-46e0-8363-ec3fed70be46.png)
![Pharo](https://user-images.githubusercontent.com/9571633/135471559-a740d913-0250-42b2-b8d1-49aad8dfe0d3.png)
![Pharo2](https://user-images.githubusercontent.com/9571633/135471576-261168d8-f998-4922-b501-06a46ffb0ca5.png)

**Version information:**
 - OS: Windows 10
 - Pharo 9 and 10 (stable and latest VM)

**Comment by guillep** [comment]
Hi @JanBliznicenko , could I ask for more details?
From the screenshot I see that the image is on a network drive.
But from your comments, you say the VM is being **executed from a network drive**.

Could you be more explicit and check some combinations?
 - what is the exact way you use to launch the whole thing? Drag n drop? Double click? command line?
 - what about other combinations (vm in non-network, image in network drive), (vm in network, image in non network drive)
 - What happens if you inspect expressions such as `FileLocator home`, `FileLocator workingDirectory`, `FileLocator imageDirectory` or `FileLocator vmDirectory`?

**Comment by JanBliznicenko** [comment]
> Hi @JanBliznicenko , could I ask for more details? From the screenshot I see that the image is on a network drive. But from your comments, you say the VM is being **executed from a network drive**.

Hello.
In case of those screenshots, both image and VM were in single directory and executed with drag and drop, so both VM and image was on network drive.

>     * what is the exact way you use to launch the whole thing? Drag n drop? Double click? command line?

I tried drag and drop, command line and double click, all with same result. 

>     * what about other combinations (vm in non-network, image in network drive), (vm in network, image in non network drive)
VM on C, image on C -> works
VM on network, image on network -> does NOT work, does NOT load .sources (argument names lost)
VM on network, image on C -> does NOT work, but loads .sources
VM on C, image on network -> works, but does NOT load .sources
...


>     * What happens if you inspect expressions such as `FileLocator home`, `FileLocator workingDirectory`, `FileLocator imageDirectory` or `FileLocator vmDirectory`?

I noticed one important thing - this network drive, specifically my home directory `\msad-fs.ms.cvut.cz\home\bliznjan`, is also mapped to X: drive. When I open `X:\Desktop\Pharo10` in Windows file explorer or terminal instead of `\\msad-fs.ms.cvut.cz\home\bliznjan\Desktop\Pharo10` (although that is the very same folder - confirmed by modifying something there) and execute Pharo from there, it fully works. So it does not seem to have problem with the drive itself, but with the path not starting with drive letter. Also, when I open FileReference to `\msad-fs.ms.cvut.cz\home\bliznjan`, I cannot see almost any files and directories (but it does show some, including some previously deleted), but when I open FileReference to `X:\`, everything works. It does not matter where is the image or VM when I open these FileReferences.

`FileLocator home` opens home directory on C drive, no matter where VM or image is located (there is actually home directory both on C drive and on network drive, but desktop, downloads and all that stuff is on network drive).
`FileLocator workingDirectory` and `FileLocator imageDirectory` opens directory with .image, but when the .image is on network drive, it has the does-not-show-almost-any-files issue described above.

Pharo Launcher works, because I installed it to path starting with X:, not the `\msad-fs.ms.cvut.cz\` notation.
